### ISSUE TITLE
Palladium caused broken Iris Shader on reload or re-enable

### ISSUE THREAD

**Issue by BruhDudeWew** [ITsMrToad/PalladiumMod#72]
### What operating systems are you seeing the problem on?

Windows

### What the loader are you seeing the problem on?

Fabric

### Describe the bug

I am using Palladium with Sodium and Iris on Minecraft 1.20.1. At first launch the shader looked fine (though my hand were glowing for some reason). Then if I either toggle on/off Iris or enter a different dimension, the shader then just completely broke. Turn it off make the game run fine. Remove Palladium make Iris run with no issues.
[latest.log](https://github.com/user-attachments/files/18184068/latest.log)

### What version of the palladium are you using?

1.0.9

**Comment by tajemniktv** [comment]
Same issue here

**Comment by ITsMrToad** [comment]
Is any particular shader causing the problem? Or any shader?

**Comment by tajemniktv** [comment]
I've only tried it on Rethinking Voxels (by Gri) and Complementary + Euphoria Patches (by Emin and EP by Spac), but it happened on both

**Comment by Fenix5fire** [comment]
> Is any particular shader causing the problem? Or any shader?

it happens with any shader. Only first launch of a shader works, then there is a chance (actually, I think that you must have no shader enabled on launch) that it will work for a few toggles, but guarantees to go black screen if you choose any other shader. With "Shrimple" it just goes insane instead of black screen. Nothing will happen if you dont switch / turn off current shader. Just tested it with a minimal setup, so it is definetely caused by Palladium itself. Also, if you just spam toggle shader button, some blocks will just stop rendering at all. Log is zipped cuz it contains 150mb worth of OpenGL error spam
[latest.zip](https://github.com/user-attachments/files/18326752/latest.zip)

**Comment by ITsMrToad** [comment]
Try updating the mod to 1.1.0, the de-duplication system has been completely rewritten there.

**Comment by BruhDudeWew** [comment]
> Try updating the mod to 1.1.0, the de-duplication system has been completely rewritten there.

Would it be possible to get a Fabric version?

**Comment by ITsMrToad** [comment]
Yes, there will be a version for fabric, after the release of ToadLib for fabric.

**Comment by Fenix5fire** [comment]
Problem is probably not solved completely. All I can tell is that it is caused by deduplication. Disabling two parameters will fix the issue and allows to fix disappearing blocks (and broken shaders) with F3+T

**Comment by ITsMrToad** [comment]
Strangely, I haven't found any errors related to reloading shaders/using them/using TACZ or DH, even on 1.20.

The spam in the log("Texture name does not refer to a texture object generated by OpenGL") is related to DH (or its incompatibility with other mods).

**Comment by 66BA9Q-ME49** [comment]
In my case, I only had to disable `Mipmap level caching` to get rid of the log spam and keep the shaders working after the reload.

**Comment by Steveplays28** [comment]
For me the issue occurs when `fastShaderPrograms` is enabled.
(Flashing and broken graphics)

**Comment by ITsMrToad** [comment]
> For me the issue occurs when fastShaderPrograms is enabled.
(Flashing and broken graphics)

Fixed(I guess...). In 1.1.7 with BSL-like/Chocapic-like shaders this definitely doesn't happen anymore.
